include/dynd/callable.hpp

```diff
@@ -155,9 +155,9 @@ namespace nd {
 
     std::intptr_t get_narg() const { return get_type()->get_npos(); }
 
-    const ndt::type &get_arg_type(std::intptr_t i) const { return get_type()->get_pos_type(i); }
+    const ndt::type get_arg_type(std::intptr_t i) const { return get_type()->get_pos_type(i); }
 
-    const array &get_arg_types() const { return get_type()->get_pos_types(); }
+    nd::array get_arg_types() const { return get_type()->get_pos_types(); }
```

Comment:
We might want to change these to return a const reference to the internal std::vector instead? Implicitly doing a copy of the whole array when get_arg_types() and other API is called feels like it adds hidden performance traps.